ios/sound.mm

```diff
@@ -25,208 +25,295 @@
 #include <AVFoundation/AVFoundation.h>
 
 
-/* Implementation *************************************************************/
-CSound::CSound ( void           (*fpNewProcessCallback) ( CVector<short>& psData, void* arg ),
-                 void*          arg,
-                 const QString& strMIDISetup,
-                 const bool     ,
-                 const QString& ) :
-    CSoundBase ( "CoreAudio iOS", fpNewProcessCallback, arg, strMIDISetup ),
-    midiInPortRef ( static_cast<MIDIPortRef> ( NULL ) )
-{
-    NSError *audioSessionError = nil;
-  
-    [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayAndRecord error:&audioSessionError];
-    [[AVAudioSession sharedInstance] requestRecordPermission:^(BOOL granted) {
-        if (granted) {
-            // ok
-        }
-    }];
-    [[AVAudioSession sharedInstance] setMode:AVAudioSessionModeMeasurement error:&audioSessionError];
-
-  //GetAvailableInOutDevices();
-}
-
-int CSound::Init ( const int /*iNewPrefMonoBufferSize*/ )
-{
+#define kOutputBus 0
+#define kInputBus 1
 
-  // store buffer size
-  //iCoreAudioBufferSizeMono = audioSession.IOBufferDuration;
 
-  // init base class
-  //CSoundBase::Init ( iCoreAudioBufferSizeMono );
-
-  // set internal buffer size value and calculate stereo buffer size
-  //iCoreAudioBufferSizeStereo = 2 * iCoreAudioBufferSizeMono;
-
-  // create memory for intermediate audio buffer
-  //vecsTmpAudioSndCrdStereo.Init ( iCoreAudioBufferSizeStereo );
-
-  return iCoreAudioBufferSizeMono;
+void checkStatus(int status){
```

Comment:
Why are these global?

Also style is wrong here.  I think clang-format is ignoring the .mm files, which it should treat like .cpp files.